lib/db.js

```diff
@@ -450,71 +387,90 @@ DB.prototype.createTable = function (reverseDomain, req) {
 
         return doCreateTables();
     })
-    .catch(function(e){console.log(e);});
 };
 
-DB.prototype._createTable = function (keyspace, schema, tableName) {
+DB.prototype._createTable = function (req, schema, tableName) {
     var self = this;
     if (!schema.attributes) {
         throw new Error('No attribute definitions for table ' + tableName);
     }
 
-    var tasks = [];
-    if (schema.secondaryIndexes) {
-        // Create secondary indexes
-        for (var idx in schema.secondaryIndexes) {
-            var indexSchema = schema.secondaryIndexes[idx];
-            tasks.push(this._createTable(keyspace, indexSchema, 'idx_' + idx +"_ever"));
+    var statics = {}, sql2;;
+    schema.index.forEach(function(elem) {
+        if (elem.type === 'static') {
+            statics[elem.attribute] = true;
         }
+    });
+
+    // Create a "_static" table if needed 
+    if (Object.keys(statics).length) {
+        sql2 = 'create table if not exists '
+        + dbu.cassID(req.keyspace + '_' + tableName + '_static') + ' (';
     }
 
-    var statics = {};
+    var hashBits = [];
+    var rangeBits = [];
+    var orderBits = [];
     schema.index.forEach(function(elem) {
-        if (elem.type === 'static') {
+        var cassName = dbu.cassID(elem.attribute);
+        if (elem.type === 'hash') {
+            hashBits.push(cassName);
             statics[elem.attribute] = true;
+        } else if (elem.type === 'range') {
+            rangeBits.push(cassName);
+            orderBits.push(cassName + ' ' + elem.order);
         }
     });
 
     // Finally, create the main data table
-    var cql = 'create table if not exists '
-        + dbu.cassID(keyspace + '_' + tableName) + ' (';
+    var sql1 = 'create table if not exists '
```

Comment:
As said earlier, I think it makes more sense to map a logical table onto a sqlite database.